openedx/core/djangoapps/heartbeat/default_checks.py

```diff
@@ -0,0 +1,89 @@
+from datetime import datetime, timedelta
+from django.core.cache import cache
+from django.db import connection
+from django.db.utils import DatabaseError
+from time import sleep, time
+from xmodule.modulestore.django import modulestore
+from xmodule.exceptions import HeartbeatFailure
+
+from .tasks import sample_task
+
+
+#DEFAULT SYSTEM CHECKS
+
+#Modulestore
+
+def check_modulestore(request):
+    # This refactoring merely delegates to the default modulestore (which if it's mixed modulestore will
+    # delegate to all configured modulestores) and a quick test of sql. A later refactoring may allow
+    # any service to register itself as participating in the heartbeat. It's important that all implementation
+    # do as little as possible but give a sound determination that they are ready.
+    try:
+        #@TODO Do we want to parse the output for split and mongo detail and return it?
+        modulestore().heartbeat()
+        return 'modulestore', True, "OK"
+    except HeartbeatFailure as fail:
+        return 'modulestore', False, unicode(fail)
+
+
+def check_database(request):
+    cursor = connection.cursor()
+    try:
+        cursor.execute("SELECT CURRENT_DATE")
+        cursor.fetchone()
+        return 'sql', True, "OK"
+    except DatabaseError as fail:
+        return 'sql', False, unicode(fail)
+
+
+#Caching
+CACHE_KEY = 'heartbeat-test'
+CACHE_VALUE = 'abc123'
+
+
+def check_cache_set(request):
+    try:
+        cache.set(CACHE_KEY, CACHE_VALUE, 30)
+        return 'cache_set', True, "OK"
+    except fail:
```

Comment:
Shouldn't this be: "except Exception as fail:" ?

Are we going to have tests that trigger the failures in each of these checks? I can see the difficulty, but it would help to catch problems like this.